pkg/smokescreen/smokescreen_test.go

```diff
@@ -625,6 +626,64 @@ func TestProxyTimeouts(t *testing.T) {
 	})
 }
 
+// TestProxyHalfClosed tests that the proxy and proxy client correctly
+// closes all connections if the proxy target attempts to half-close
+// the TCP connection.
+func TestProxyHalfClosed(t *testing.T) {
```

Comment:
I like this test! What do you think about adding a second test for a proxy _client_ closing its write socket? That would cover both cases and make sure we don't end up with regressions.